asv/benchmarks.py

```diff
@@ -120,6 +120,10 @@ def log_result(msg):
         result['started_at'] = datetime.datetime.utcnow()
 
         for param_idx, params in param_iter:
+            if benchmark['params'] and param_idx not in benchmark['_selected_idx']:
```

Comment:
I'd maybe prefer a new `selected_idx` keyword argument to `run_benchmark` instead of storing the selection mask in the benchmark dicts.

Then it's just a matter of adding `selected_idx=self._benchmark_selection[name]` here
https://github.com/philpep/asv/blob/2df016ddbd6236c1e6ddbfd94c366198702f2b80/asv/benchmarks.py#L626-L630